### ISSUE TITLE
Add nav to the selectable elements for the group block

### ISSUE THREAD

**Issue by carolinan** [WordPress/gutenberg#67407]
## What problem does this address?

I am splitting this out of https://github.com/WordPress/gutenberg/pull/57664

There seems to be a consensus that the `nav` HTML element should be added to the list of selectable elements in the Advanced section of the settings for the group block. Currently, the tag can only be added with code.
The purpose of the change would be to make it easier for advanced users who are not developers to create accessible navigations.

Even if there is consensus among contributors, there is still one important aspect that must to be solved:
When multiple nav elements exists on a page, the user needs to be able to distinguish the different nav elements, or, in other words, 
users needs to know the purpose of the navigation.
The main navigation, the next and previous posts in a query, the comment pagination etc. and any custom navigations that the users adds, they need to be identifiable.

In the linked PR for the post navigation block, this is solved by adding a block attribute and an option where the user can customize the aria-label for the nav element:

```
<InspectorControls group="advanced">
				<TextControl
					__next40pxDefaultSize
					__nextHasNoMarginBottom
					label={ __( 'Navigation name' ) }
					value={ ariaLabel || __( 'Posts' ) }
					onChange={ ( newLabel ) =>
						setAttributes( { ariaLabel: newLabel } )
					}
				/>
			</InspectorControls>
```

How should this be solved for the group block?

-----
[WCAG: Using ARIA landmarks to identify regions of a page](https://www.w3.org/WAI/WCAG22/Techniques/aria/ARIA11)
[WordPress.org Theme requirement: ARIA Landmark Roles](https://make.wordpress.org/themes/handbook/review/accessibility/required/#aria-landmark-roles)

**Comment by carolinan** [comment]
@fabiankaegy @t-hamano @draganescu @talldan

**Comment by t-hamano** [comment]
The Group block [has `ariaLabel` block support](https://github.com/WordPress/gutenberg/blob/8acc11dec8cbb3e335002c1119da1d5b6b90eabc/packages/block-library/src/group/block.json#L29), but this block support doesn't provide a UI. What if we add the control for `ariaLabel` only if `tagName` is `nav`?

I created a branch to test this: https://github.com/WordPress/gutenberg/compare/group-aria-label?expand=1

https://github.com/user-attachments/assets/ea0a31aa-ab2d-48c6-933b-c414de648582

**Comment by carolinan** [comment]
@t-hamano I would support such a change to the group block.

**Comment by bacoords** [comment]
+1 In support of this. There should be the ability to put lists and other blocks inside a `nav` element without relying on the core navigation block or a custom block

**Comment by afercia** [comment]
> When multiple nav elements exists on a page, the user needs to be able to distinguish the different nav elements,

To illustrate, and also to clarify the goal of the related PR at https://github.com/WordPress/gutenberg/pull/68611, `<nav>` elements are also ARIA landmarks and listed by assistive technology to allow users to quickly jump to them by using dedicated keyboard shortcuts or other tools. For example, VoiceOver lists all the landmark in a page in the related panel in the 'rotor'. When multiple navigation elements are in a page and are _not_ labeled, it's impossible to understand the scope of a navigation. Screenshot:

![Image](https://github.com/user-attachments/assets/63ec9ea9-8310-4f99-bdb8-f14a46db8b66)
